src/paho/mqtt/client.py

```diff
@@ -3328,16 +3338,44 @@ def _handle_publish(self):
             return MQTT_ERR_SUCCESS
         elif message.qos == 1:
             self._handle_on_message(message)
-            return self._send_puback(message.mid)
+            if self._manual_ack:
+                return MQTT_ERR_SUCCESS
+            else:
+                return self._send_puback(message.mid)
         elif message.qos == 2:
+
             rc = self._send_pubrec(message.mid)
+
             message.state = mqtt_ms_wait_for_pubrel
             with self._in_message_mutex:
                 self._in_messages[message.mid] = message
+
             return rc
         else:
             return MQTT_ERR_PROTOCOL
 
+    def ack( self, mid, qos ):
```

Comment:
```suggestion
    def ack(self, mid: int, qos: int) -> int:
```

The formatting isn't the expected one (space around parenthesis) and we if could add types it's better.

None are required, because both linter (black for formating and mypy for typing) runs (`tox -e lint`) but only give warning. But if the PR could avoid increasing errors it will be less work in the future :)